src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -396,40 +373,41 @@ case class AppDefinition(
         for {
           launched <- task.launched
           effectiveIpAddress <- task.effectiveIpAddress(this)
-        } yield appPorts.zip(launched.hostPorts).zipWithIndex.map {
-          case ((appPort, hostPort), portIndex) =>
-            PortAssignment(Some(appPort.name), portIndex, effectiveIpAddress, hostPort)
+        } yield appPorts.zip(launched.hostPorts).map {
+          case (appPort, hostPort) =>
+            PortAssignment(Some(appPort.name), effectiveIpAddress, hostPort)
         }.toList
     }
 
     def fromPortMappings: Option[Seq[PortAssignment]] =
       for {
-        pms <- portMappings
+        c <- container
+        pms <- c.portMappings
```

Comment:
no idea if this is correct, but can you do "if pms.hostPort.isDefined" here and remove it from the filter? If not, line 387 should be "withFilter()" instead of "filter()"